qcodes/tests/drivers/test_keithley_s46.py

```diff
@@ -0,0 +1,156 @@
+import pytest
+
+from qcodes.instrument_drivers.tektronix.Keithley_s46 import (
+    S46, LockAcquisitionError
+)
+
+import qcodes.instrument.sims as sims
+visalib = sims.__file__.replace('__init__.py', 'Keithley_s46.yaml@sim')
+
+
+def test_aliases_dict():
+    """
+    Test the class attribute 'aliases' which maps channel aliases
+    (e.g. A1, B2, etc) to channel numbers.
+    """
+    def calc_channel_nr(alias: str) -> int:
+        """
+        We perform the calculation in a different way to verify correctness
+        """
+        offset_dict = dict(zip(["A", "B", "C", "D", "R"], range(0, 32, 6)))
+        return offset_dict[alias[0]] + int(alias[1:])
+
+    assert all([nr == calc_channel_nr(al) for al, nr in S46.aliases.items()])
+
+
+class S46LoggedAsk(S46):
+    """
+    A version of the driver which logs every ask command. We need this to
+    assert that ':CLOS?' is called once during initialization
+    """
+    def __init__(self, *args, **kwargs):
+        self._ask_log = []
+        super().__init__(*args, **kwargs)
+
+    def ask(self, cmd: str):
+        self._ask_log.append(cmd)
```

Comment:
You could probably do this with the existing logging in the instrument and pytests log capturing functions